lib/archethic_web/controllers/faucet_controller.ex

```diff
@@ -0,0 +1,80 @@
+defmodule ArchEthicWeb.FaucetController do
+  @moduledoc false
+
+  alias ArchEthic.TransactionChain.{
+    Transaction,
+    TransactionData,
+    TransactionData.Ledger,
+    TransactionData.UCOLedger
+  }
+
+  alias ArchEthic.Crypto
+
+  use ArchEthicWeb, :controller
+
+  def index(conn, __params) do
+    conn
+    |> put_resp_header("cache-control", "no-cache, no-store, must-revalidate")
+    |> put_resp_header("pragma", "no-cache")
+    |> put_resp_header("expires", "0")
+    |> render("index.html", address: "")
+  end
+
+  def create_transfer(conn, %{"address" => address}) do
+    case transfer(address, Crypto.default_curve()) do
```

Comment:
I think you need to check if the address is well formatted

Reply:
Is there any utility API function to do that or do I have to pattern match the binary explicitly?

Reply:
You can take inspiration from https://github.com/archethic-foundation/archethic-node/blob/39e466dab1feb26c6446ba248247cbd58354f93b/lib/archethic_web/controllers/explorer_controller.ex#L13

Reply:
I shouldn't do this check right?

`{:ok, tx} <- ArchEthic.search_transaction(address)`

Because it won't exist for first time users.

Reply:
Yes indeed for the first time address it will not exist. So no need to make this check